control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1411,6 +1412,43 @@ func (r *reconciler) reconcileCloudCredentialSecrets(ctx context.Context, hcp *h
 		}); err != nil {
 			errs = append(errs, fmt.Errorf("failed to reconcile csi driver secret: %w", err))
 		}
+	case hyperv1.OpenStackPlatform:
+		credentialsSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: hcp.Namespace, Name: hcp.Spec.Platform.OpenStack.IdentityRef.Name}}
+		if err := r.cpClient.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), credentialsSecret); err != nil {
+			return []error{fmt.Errorf("failed to get cloud credentials secret in hcp namespace: %w", err)}
+		}
+		caCertData := openstack.GetCACertFromCredentialsSecret(credentialsSecret)
+		cloudName := hcp.Spec.Platform.OpenStack.IdentityRef.CloudName
+
+		csiCredentialSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: "openshift-cluster-csi-drivers", Name: "openstack-cloud-credentials"}}
```

Comment:
This code block from line 1423 -1428 is repeated a couple of times, the only thing that changes is the name and namespace of the secret used. It might be worth to extract it to a function.

Reply:
excellent idea. Done